specreduce/extract.py

```diff
@@ -404,6 +432,23 @@ def __call__(self, image=None, trace_object=None,
             # replace the variances to avoid a divide by zero warning
             variance[variance == 0] = np.nan
 
+        # If the trace is not flat, shift the rows in each column
+        # so the image is aligned along the trace:
+        if isinstance(trace_object, FlatTrace):
+            mean_init_guess = trace_object.trace
+        else:
+            img = _align_along_trace(
+                img,
+                trace_object.trace,
+                disp_axis=disp_axis,
+                crossdisp_axis=crossdisp_axis
+            )
+            # Choose the initial guess for the mean of
+            # the Gaussian profile:
+            mean_init_guess = np.broadcast_to(
+                img.shape[crossdisp_axis] // 2, img.shape[disp_axis]
+            )
```

Comment:
would just swapping this out with the "remainder" from casting `trace_object.trace` to their integer rolls be enough to correct the means to the actual trace position?

Reply:
I think it would improve the initial guess for the mean parameter given to the Gaussian spatial profile fit by <1 pixel.